qcfractal/interface/models/rest_models.py

```diff
@@ -1122,3 +1140,115 @@ class QueueManagerPUTResponse(ProtoModel):
 
 
 register_model("queue_manager", "PUT", QueueManagerPUTBody, QueueManagerPUTResponse)
+
+
+## advanced procedures queries
+
+class OptimizationBestResultsBody(ProtoModel):
+    class Data(ProtoModel):
+        opt_ids: QueryObjectId = Schema(
+            None,
+            description="List of optimization procedure Ids to fetch their BEST results from the database."
+        )
+
+    # TODO: not yet supported
+    meta: QueryMetaProjection = Schema(
+        QueryMetaProjection(),
+        description=common_docs[QueryMetaProjection]
+    )
+    data: Data = Schema(
+        ...,
+        description="The keys with data to search the database on for Procedures."
+    )
+
+class OptimizationAllResultsBody(ProtoModel):
+    class Data(ProtoModel):
+        opt_ids: QueryObjectId = Schema(
+            None,
+            description="List of optimization procedure Ids to fetch their ALL their results from the database."
+        )
+
+    # TODO: not yet supported
+    meta: QueryMetaProjection = Schema(
+        QueryMetaProjection(),
+        description=common_docs[QueryMetaProjection]
+    )
+    data: Data = Schema(
+        ...,
+        description="The keys with data to search the database on for Procedures."
+    )
+
+class OptimizationInitialMoleculesBody(ProtoModel):
+    class Data(ProtoModel):
+        opt_ids: QueryObjectId = Schema(
+            None,
+            description="List of optimization procedure Ids to fetch their initial  molecules from the database."
+        )
+
+    # TODO: not yet supported
+    meta: QueryMetaProjection = Schema(
+        QueryMetaProjection(),
+        description=common_docs[QueryMetaProjection]
+    )
+    data: Data = Schema(
+        ...,
+        description="The keys with data to search the database on for Procedures."
+    )
+
+class OptimizationFinalMoleculesBody(ProtoModel):
```

Comment:
We usually try to keep plurality here `OptimizationInitialMoleculesBody` -> `OptimizationInitialMoleculeBody`. Same for `OptimizationBestResultsBody` -> `OptimizationFinalResultBody`.

Reply:
I think Optimization being singular is good and following the rule, but the second part (i.e Molecules) indicates that the return per opt is a list of associated objects. And overall, the class applies for a list of optimizations, returning list of Molecules.

Reply:
Hmm, I always thought of it as a "Molecule per id" not "Molecule". Minor point.

Reply:
Ok, I changed them to singular.